next/13/app-directory-boilerplate/index.ts

```diff
@@ -599,62 +599,76 @@ const updateLayoutComponent = (sourceFile: SourceFile) => {
 	}`);
 };
 
-const extractStatements = (sourceFile: SourceFile) => {
-	const statements = sourceFile.getStatements();
+const findComponent = (sourceFile: SourceFile) => {
+	const defaultExportedFunctionDeclaration = sourceFile
+		.getFunctions()
+		.find((f) => f.isDefaultExport());
+
+	if (defaultExportedFunctionDeclaration !== undefined) {
+		return defaultExportedFunctionDeclaration;
+	}
+
+	const exportAssignment = sourceFile
+		.getStatements()
+		.find((s) => Node.isExportAssignment(s));
 
-	const exportAssignment = statements.find((s) => Node.isExportAssignment(s));
-	const exportedIdentifierName = exportAssignment
-		?.getFirstDescendantByKind(SyntaxKind.Identifier)
-		?.getText();
+	const declarations =
+		exportAssignment
+			?.getFirstDescendantByKind(SyntaxKind.Identifier)
+			?.getSymbol()
+			?.getDeclarations() ?? [];
 
 	let component:
 		| ArrowFunction
 		| FunctionExpression
 		| FunctionDeclaration
 		| undefined;
 
-	const functionDeclarations: FunctionDeclaration[] = [];
-	const importDeclarations: ImportDeclaration[] = [];
-	const variableStatements: VariableStatement[] = [];
-
-	statements.forEach((s) => {
-		if (Node.isExportAssignment(s)) {
-			return;
-		}
-
-		if (Node.isImportDeclaration(s)) {
-			importDeclarations.push(s);
-		}
-
-		if (Node.isVariableStatement(s)) {
-			const declaration = s.getFirstDescendantByKind(
-				SyntaxKind.VariableDeclaration,
-			);
-			const initializer = declaration?.getInitializer();
+	declarations.forEach((d) => {
+		if (Node.isVariableDeclaration(d)) {
+			const initializer = d?.getInitializer();
 
 			if (
-				declaration?.getName() === exportedIdentifierName &&
-				(Node.isArrowFunction(initializer) ||
-					Node.isFunctionExpression(initializer))
+				Node.isArrowFunction(initializer) ||
+				Node.isFunctionExpression(initializer)
 			) {
 				component = initializer;
 				return;
 			}
+		}
 
-			variableStatements.push(s);
+		if (Node.isFunctionDeclaration(d)) {
+			component = d;
 		}
+	});
 
-		if (Node.isFunctionDeclaration(s)) {
-			if (s.isDefaultExport() || s.getName() === exportedIdentifierName) {
-				component = s;
-				return;
-			}
+	return component ?? null;
+};
 
-			functionDeclarations.push(s);
-		}
+const buildLayoutClientComponentFromUnderscoreApp = (
+	sourceFile: SourceFile,
+) => {
+	const component = findComponent(sourceFile);
+
+	if (component === null) {
+		return;
+	}
+
+	if (!Node.isArrowFunction(component)) {
+		component.rename('LayoutClientComponent');
```

Comment:
can you explain this line?

Reply:
im taking _app file and based on it i build layout client component, so i want to change original _app component name to more appropriate